software/generic/Effect.cpp

```diff
@@ -1,10 +1,6 @@
 #include "Effect.hpp"
 
-Effect::Effect(uint8_t numLeds)
-    : numLeds(numLeds), deviceType(DeviceType::Wearable) {}
-
-Effect::Effect(uint8_t numLeds, DeviceType deviceType)
-    : numLeds(numLeds), deviceType(deviceType) {}
+Effect::Effect(DeviceDescription *const device): device(device) {}
```

Comment:
I think this can be `DeviceDescription const *const device)`

Reply:
Should it be? Aren't pointers passed by value in which case `const`ness doesn't apply?

Reply:
Yes, you are correct.